File: cpp/bc1049.cpp
#include <iostream>

using namespace std;

int main()
{
	string fVer, sVer, tVer;

	cin >> fVer >> sVer >> tVer;
	
	if(fVer == "vertebrado")
	{
		if(sVer == "ave")
			tVer == "carnivoro"? cout << "aguia" << endl : cout << "pomba" << endl;
		else
			tVer == "onivoro"? cout << "homem" << endl : cout << "vaca" << endl;
	}
	else
	{
		if(sVer == "inseto")
			tVer == "hematofago"? cout << "pulga" << endl : cout << "lagarta" << endl;
		else
			tVer == "hematofago"? cout << "sanguessuga" << endl : cout << "minhoca" << endl;
	}
}

File: python/bc1049.py
fDe = input()
sDe = input()
tDe = input()

if fDe == "vertebrado":
    if sDe == "ave":
        res = "aguia" if tDe == "carnivoro" else "pomba"
    else:
        res = "homem" if tDe == "onivoro" else "vaca"
else:
    if sDe == "inseto":
        res = "pulga" if tDe == "hematofago" else "lagarta"
    else:
        res = "sanguessuga" if tDe == "hematofago" else "minhoca"

print(res)